src/sfctl/tests/help_text_test.py

```diff
@@ -0,0 +1,364 @@
+            method will take care of that.
+
+        subgroups (tuple of strings): This represents all of the subgroups expected in the
+            help text. This tuple must be in alphabetical order.
+
+        commands (tuple of strings): This represents all of the commands expected in the
+            help text. This tuple must be in alphabetical order.
+
+        Help text has one of the two following formats:
+
+        1.  Group
+                sfctl chaos: Start, stop and report on the chaos test service.
+
+            Subgroups:
+                chaos      : Start, stop and report on the chaos test service.
+                cluster    : Select, manage and operate Service Fabric clusters.
+
+            Commands: [This section may not exist depending on input]
+                get     : Gets something.
+                events  : Gets the next segment of the Chaos events...
+                      range.
+
+        or
+
+        2.  Command
+                sfctl application create: Creates a Service Fabric appli...
+
+            Arguments
+                --app-version [Required]: The version of the application type as defined in the
+                                          manifest.
+                --timeout -t            : Server timeout in seconds.  Default: 60.
+
+            Global Arguments
+                --debug                 : Increase logging verbosity to show all debug logs.
+        """
+
+        try:
+            # This variable tracks what sections of the help text we are in
+            # Possibilities are Group, Subgroups, Commands, Command, Arguments,
+            # and Global Arguments.
+            # Once we no longer support python 2, change section options of enums
+            section = 'Start'
+
+            # A tracker to know which subgroup or command has appeared in help text
+            subgroups_index = 0
+            commands_index = 0
+
+            help_command = command_input + ' -h'
+
+            # Call the provided command in command line
+            # Do not split the help_command, as that breaks behavior: Linux ignores the splits.
+            pipe = Popen(help_command, shell=True, stdout=PIPE, stderr=PIPE)
+            # returned_string and err are returned as bytes
+            (returned_string, err) = pipe.communicate()
+
+            # If there is an error message, then assert
+            if err:
+                err = err.decode('utf-8')
+                print(err, file=stderr)
+                self.assertEqual(b'', err, msg='ERROR: in command: ' + help_command)
+
+            # If there is no help text, then assert
+            if not returned_string:
+                self.fail('No help text in command: ' + help_command)
+
+            returned_string = returned_string.decode('utf-8')
+            lines = returned_string.splitlines()
+
+            for line in lines:
+
+                print(line, file=stderr)
```

Comment:
Why printing to stderr?

Reply:
Printing to stdout doesn't get added to the cmdline screen. I think it's caught and printed together at the end of the test, but for debugging and seeing the progress of the test I found this to be much easier.